core/src/server/init/CpuChecker.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#include "server/init/CpuChecker.h"
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "index/thirdparty/faiss/FaissHook.h"
+#include "index/thirdparty/faiss/utils/instruction_set.h"
+#include "utils/Log.h"
+#include "utils/StringHelpFunctions.h"
+
+namespace milvus {
+namespace server {
+
+Status
+CpuChecker::CheckCpuInstructionSet() {
+    std::vector<std::string> instruction_sets;
+
+    auto& instruction_set_inst = faiss::InstructionSet::GetInstance();
+    if (faiss::support_avx512()) {
+        instruction_sets.emplace_back("avx512");
+    }
+    if (instruction_set_inst.AVX2()) {
+        instruction_sets.emplace_back("avx2");
+    }
+    if (instruction_set_inst.SSE42()) {
+        instruction_sets.emplace_back("sse4_2");
+    }
+
+    if (instruction_sets.empty()) {
+        std::string msg =
+            "CPU instruction sets are not supported. Ensure the CPU supports at least one of the following instruction "
```

Comment:
If one CPU support AVX512, it must support AVX2;
if one CPU support AVX2, it must also support SSE4_2.
I think only check SSE4_2 is fine.